File: Leetcode/1869. Longer Contiguous Segments of Ones than Zeros/solution.py
class Solution:
    def checkZeroOnes(self, s: str) -> bool:
        max_zero = 0
        max_ones = 0
        temp = 0

        for i in range(len(s)):
            if i > 0 and i < len(s) and s[i] != s[i-1]:
                temp = 0

            temp += 1

            max_zero = max(temp, max_zero) if s[i] == '0' else max_zero
            max_ones = max(temp, max_ones) if s[i] == '1' else max_ones

        return True if max_ones > max_zero else False

File: Leetcode/1869. Longer Contiguous Segments of Ones than Zeros/solution.rs
use std::cmp;

impl Solution {
    pub fn check_zero_ones(s: String) -> bool {
        let mut max_zero = 0;
        let mut max_ones = 0;
        let mut temp = 0;

        for i in 0..s.len() {
            if i > 0 && i < s.len() && s.as_bytes()[i] != s.as_bytes()[i - 1] {
                temp = 0;
            }

            temp += 1;
            max_zero = if s.chars().nth(i).unwrap() == '0' {
                cmp::max(max_zero, temp)
            } else {
                max_zero
            };
            max_ones = if s.chars().nth(i).unwrap() == '1' {
                cmp::max(max_ones, temp)
            } else {
                max_ones
            };
        }

        return if max_ones > max_zero { true } else { false };
    }
}
